### PR TITLE
MISUV-204 - Remove the old subscription route, made property models a…

### PR DESCRIPTION
**Pull request by AlexRimmerHMRC**
…ccounting method mandatory

### CODE REVIEW

File: test/services/RegistrationServiceSpec.scala [hunk 1]
```diff
@@ -1,59 +0,0 @@
-/*
- * Copyright 2020 HM Revenue & Customs
```
Comment (JSimmons072019) [inline test/services/RegistrationServiceSpec.scala:2]: should this really be deleted

File: test/connectors/RegistrationConnectorSpec.scala [hunk 2]
```diff
@@ -1,136 +0,0 @@
-/*
- * Copyright 2020 HM Revenue & Customs
```
Comment (JSimmons072019) [inline test/connectors/RegistrationConnectorSpec.scala:2]: should this really be deleted

File: it/helpers/servicemocks/RegistrationStub.scala [hunk 3]
```diff
@@ -33,13 +33,4 @@ object RegistrationStub extends WireMockMethods {
       .thenReturn(status = responseStatus, body = responseBody)
   }
 
-  @Deprecated
```
Comment (JSimmons072019) [inline it/helpers/servicemocks/RegistrationStub.scala:36]: will this be affected by the other deletes

File: app/models/subscription/incomesource/SignUpRequest.scala [hunk 4]
```diff
@@ -33,7 +32,7 @@ case class BusinessIncomeModel(tradingName: Option[String],
                                accountingPeriod: AccountingPeriod,
                                accountingMethod: AccountingMethod)
 
-case class PropertyIncomeModel(accountingMethod: Option[AccountingMethod]) //TODO change to non option when cash and accruals has been added to property
```
Comment (JSimmons072019) [inline app/models/subscription/incomesource/SignUpRequest.scala:36]: it's still possible to have a option in front-end if a feature switch isn't on

File: test/models/incomesource/SignUpRequestSpec.scala [hunk 5]
```diff
@@ -41,23 +39,20 @@ class SignUpRequestSpec extends UnitSpec {
       AccountingPeriod(startDate = DateModel(startDay, startMonth, startYear), endDate = DateModel(endDay, endMonth, endYear)),
       AccountingMethod.feCash)
 
-    val propertyIncome = PropertyIncomeModel(Some(Cash))
-    val propertyNoIncome = PropertyIncomeModel(None)
+    val propertyIncome = PropertyIncomeModel(Cash)
 
-    def businessIncomeJson(testArn: Option[String] = None) = Json.obj(
-      "businessDetails" ->  Seq(Json.obj(
+    def businessIncomeJson(testArn: Option[String] = None): JsObject = Json.obj(
```
Comment (JSimmons072019) [inline test/models/incomesource/SignUpRequestSpec.scala]: Optional?

File: test/services/mocks/TestRegistrationService.scala [hunk 6]
```diff
@@ -1,53 +0,0 @@
-/*
```
Comment (JSimmons072019) [inline test/services/mocks/TestRegistrationService.scala:1]: reg deletion?

File: test/utils/TestConstants.scala [hunk 7]
```diff
@@ -327,7 +300,7 @@ object TestConstants {
     arn = None,
     businessIncome = None,
     propertyIncome = Some(PropertyIncomeModel(
-      accountingMethod = Some(Cash)
```
Comment (JSimmons072019) [inline test/utils/TestConstants.scala:330]: still needs to be optional

File: test/connectors/mocks/TestRegistrationConnector.scala [hunk 8]
```diff
@@ -1,90 +0,0 @@
-/*
- * Copyright 2020 HM Revenue & Customs
```
Comment (JSimmons072019) [inline test/connectors/mocks/TestRegistrationConnector.scala:2]: Reg deletion?

File: app/services/RegistrationService.scala [hunk 9]
```diff
@@ -1,58 +0,0 @@
-/*
```
Comment (JSimmons072019) [inline app/services/RegistrationService.scala:1]: Reg deletion?

**Comment by jameshforster** [comment]
Coverage is below minimum

**Comment by AlexRimmerHMRC** [comment]
@jameshforster Coverage is now above minimum

**Pull request merged**